src/v2/Utils/Hooks/__tests__/useFadeTransition.jest.tsx

```diff
@@ -0,0 +1,63 @@
+import { render } from "@testing-library/react"
+import { renderHook } from "@testing-library/react-hooks"
+import { useFadeTransition } from "../useFadeTransition"
+
+jest.mock("../../wait", () => ({
+  wait: () => Promise.resolve(),
+}))
+
+describe("useFadeTransition", () => {
+  it("sets some defaults", () => {
+    const { result } = renderHook(() => useFadeTransition())
+
+    expect(result.current.mode).toEqual("Resting")
+    expect(result.current.status).toEqual("In")
+  })
+
+  it("registers elements", () => {
+    const { result } = renderHook(() => useFadeTransition())
+
+    const Example = () => {
+      return (
+        <div>
+          <div ref={result.current.register(0)}>one</div>
+          <div ref={result.current.register(2)}>three</div>
+          <div ref={result.current.register(1)}>two</div>
+        </div>
+      )
+    }
+
+    render(<Example />)
+
+    expect(
+      result.current.refs.current.map(ref => ref.current!.innerHTML)
+    ).toEqual(["one", "two", "three"])
```

Comment:
So I think this demonstrates that it doesn't matter which order you call `register` it'll respect the position passed in, right?